### PR TITLE
docs: Add docstring code samples for `Series.apply` and `DataFrame.map`

### PR DESCRIPTION
**Pull request by shobsi**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 1]
```diff
@@ -2159,8 +2159,68 @@ def map(self, func, na_action: Optional[str] = None) -> DataFrame:
            In pandas 2.1.0, DataFrame.applymap is deprecated and renamed to
            DataFrame.map.
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            Let's use ``reuse=False`` flag to make sure a new ``remote_function``
```
Comment (tswast) [inline third_party/bigframes_vendored/pandas/core/frame.py:2167]: Only the code sample itself should be indented 4 spaces. That's so we can trick Cloud RAD into using a preformatted text font. This text doesn't / shouldn't be in that code-style font.

Comment (shobsi) [reply third_party/bigframes_vendored/pandas/core/frame.py]: Thanks for the catch, fixed!

**Comment by shobsi** [comment]
Appearance of the docs after staging the changes on cloud.devsite:

https://screenshot.googleplex.com/4UgUVJ6iBiLRcEC
https://screenshot.googleplex.com/4rZVpuxjGZiDnqh
https://screenshot.googleplex.com/Bmx4wbE9Xh6myrF
https://screenshot.googleplex.com/B784QnV3rVN7uXn

**Review by ashleyxuu — APPROVED** [review]

**Comment by ashleyxuu** [comment]
Thanks for doing the staging :) Glad you fix the error.

**Pull request merged**